Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/EntityProxyDataStream.cs

```diff
@@ -168,7 +169,10 @@ public DataStreamPendingWriter<TInstance> CreateDataStreamPendingWriter()
 
         public DataStreamActiveReader<TInstance> CreateDataStreamActiveReader()
         {
-            return new DataStreamActiveReader<TInstance>(m_ActiveArrayData.DeferredJobArray);
+            bool isDeferredRequired = !m_ActiveArrayData.GetDependency(AccessType.SharedRead).IsCompleted;
```

Comment:
Maybe just a quick comment here as to why SharedRead and why that means we need to use the deferred?

Reply:
Good call, Added.